components/FishForm.js

```diff
@@ -0,0 +1,124 @@
+import React, { Component } from 'react';
+import { StyleSheet, Text, View, TextInput, TouchableOpacity } from 'react-native';
+import { updateFishEntry } from '../actions'
+import { connect } from "react-redux";
+import RNPickerSelect from 'react-native-picker-select';
+
+class FishForm extends Component {
+  state = {
+    species: '',
+    image: '',
+    length: '',
+    weight: '',
+		location: '',
+		flyName: ''
+  }
+
+  handleSpecies = (text) => {
+    this.setState({ species: text })
+    this.props.updateFishEntry('species', text)
+  }
+
+  handleImage = (text) => {
+		this.setState({ image: text })
+		this.props.updateFishEntry('image', text)
+  }
+
+  handleLength = (text) => {
+		this.setState({ length: text })
+		this.props.updateFishEntry('length', parseInt(text))
+  }
+
+  handleWeight = (text) => {
+		this.setState({ weight: text })
+		this.props.updateFishEntry('weight', parseInt(text))
+  }
+
+  handleLocation = (text) => {
+		this.setState({ location: text })
+		this.props.updateFishEntry('location', text)
+	}
+	
+	handleFlyId = (value) => {
+		this.setState({ flyName: value })
+		this.props.updateFishEntry('fly_id', parseInt(value))
+	}
```

Comment:
Similar to the fly form, we will want to refactor these functions to DRY up the code.

Reply:
Yes. I agree. Since event targeting is not a thing, I hope there is another way to refactor this in React Native.